### DISCUSSION TITLE
AutoScalingListener is created but remains OutOfSync

### DISCUSSION THREAD

**Discussion by hhami-newday** [community/community#160678] [Actions] [answered]
Hello! I am fairly new to ArgoCD and k8s so maybe it is a silly question. 

I have created a multi-source application in ArgoCD which builds the [controller](https://github.com/actions/actions-runner-controller/tree/master/charts/gha-runner-scale-set-controller) and the [scale-set](https://github.com/actions/actions-runner-controller/tree/master/charts/gha-runner-scale-set). In Github, I can queue a workflow using the correct `runs-on` label and it all works as expected.

However in ArgoCD, the controller (I am assuming) creates an autoscalinglistener (and a role / rolebinding resource) at the top-level of the application. 

This means that my ArgoCD sync always shows as OutOfSync as the desired manifest is blank but live manifest is obviously populated. Have I done something wrong here? 

Side note: The listener pod running under the autoscalinglistener never reaches healthy state. It is always 'progressing' but the app itself seems to work and run my actions accordingly.

Any guidance is appreciated!

**Comment by luis-lf** [comment]
I have the same issue, everything seems to work fine but argo shows that the scale-set app is out of sync with this message for the listener role and role-binding:

```
 OutOfSync (This resource is not present in the application's source. It will be deleted from Kubernetes if the prune option is enabled during sync.)
```

Were you able to find a solution for this?

**Reply by hhami-newday** [reply]
I wasn't. I was hoping someone could shed some light on it. 

You can do some [resource exclusions in ArgoCD](https://argo-cd.readthedocs.io/en/stable/operator-manual/declarative-setup/#resource-exclusioninclusion) but currently my implementation shares a namespace with some other applications and I think the exclusions cannot be granular enough for me to exclude just these particular components for this application. I believe it would affect other kinds (roles, for example) in the same namespace which isn't what I want. 

If you have it in a completely isolated namespace, maybe that will work for you.

**Reply by juris** [reply]
Same here. scale-set-controller:0.9.3 ArgoCD:2.12.0

**Reply by AurimasNav** [reply]
> I wasn't. I was hoping someone could shed some light on it.
> 
> You can do some [resource exclusions in ArgoCD](https://argo-cd.readthedocs.io/en/stable/operator-manual/declarative-setup/#resource-exclusioninclusion) but currently my implementation shares a namespace with some other applications and I think the exclusions cannot be granular enough for me to exclude just these particular components for this application. I believe it would affect other kinds (roles, for example) in the same namespace which isn't what I want.
> 
> If you have it in a completely isolated namespace, maybe that will work for you.

It does not seem to have namespace parameter so it is cluster wide.

**Reply by SPodjasek** [reply]
See this comment: https://github.com/actions/actions-runner-controller/issues/3533#issuecomment-2232996941

**Comment by juris** [comment]
[Found](https://github.com/actions/actions-runner-controller/issues/3533) the related issue

**Reply by SPodjasek** [reply, marked as answer]
And this comment should be referenced as an resolution:

https://github.com/actions/actions-runner-controller/issues/3533#issuecomment-2232996941
